doc/decisions/internal_cache.md

```diff
@@ -2,26 +2,98 @@
 
 ## Problem
 
-When doing kdbGet() possible more keys are returned which might be
-confusing. When doing a second kdbGet() with a new keyset
-no keys might be returned, because it is up-to-date.
+`kdbGet` might return more or less keys than requested.
+This was found confusing several times.
 
-When doing kdbSet() a deep duplication is needed.
+### More Keys
 
-Idea: keep a duplicated keyset internally. Return (non-deep?)
-duplications exactly of the parts as requested.
+Even if calling `kdbGet` with a parent key below a mountpoint, `kdbGet` will nevertheless return all keys of the mountpoint.
+Pseudo code example, assuming there is a mountpoint at `/mountpoint` and a key `/mountpoint/other`:
+
+```
+kdbGet (kdb, ks, keyNew("/mountpoint/below"));
+assert (ksLookup (ks, "/mountpoint/other") == NULL);
+```
+
+If was found unexpected that this assert will fail.
+
+### Less Keys
+
+When doing a second `kdbGet` with a new keyset no keys might be returned, because kdb internally is up-to-date.
+Pseudo code example, assuming there is a key `/somewhere/key`:
+
+```
+kdbGet (kdb, ks1, keyNew("/somewhere"));
+assert (ksLookup (ks1, "/somewhere/key") != NULL);
+kdbGet (kdb, ks2, keyNew("/somewhere"));
+assert (ksLookup (ks2, "/somewhere/key") != NULL);
+```
+
+If was found unexpected that the second assert will fail.
```

Comment:
Oh yeah, of course. `kdbGet` never reaches `backendsMerge`. However, I think this would work, if at least one of the backends reports changed data. So for example, if the value of `/somewhere/key` changes then things should be fine.

So I think we need a clarification here:

```
When doing a second `kdbGet` with a new keyset no keys will be returned when no backends report changed data, because kdb internally is thinks the data is already up-to-date.
```

In my mind, this just is one more point in favour of going with the option of `kdbData (kdb)` and removing the `KeySet *` argument from `kdbGet` and `kdbSet`.

Reply:
> When doing a second `kdbGet` with a new keyset no keys will be returned when no backends report changed data

This is horrible. Why would the user care if it was updated or not? If I do a `kdbGet` I want the current data, regardless of whether I already got it before or not.

Reply:
No I meant we need a clarification in this decision. 

```patch
- When doing a second `kdbGet` with a new keyset no keys might be returned, because kdb internally is up-to-date.
+ When doing a second `kdbGet` with a new keyset no keys will be returned when no backends report changed data, because kdb internally thinks the data is already up-to-date.
```

> **Note**: Can't make this a proper suggestion (in this thread) because there is a <code>&#96;&#96;&#96;</code> in the code region of this comment thread.

This of course a horrible problem. Which is why I'm now more in favour of the `kdbData(kdb)` option. And in fact this is a more general reason (than change tracking) to clarify the allowed sequences of `kdbGet`/`kdbSet` and their meaning.